addons/l10n_es_edi_tbai/models/account_edi_format.py

```diff
@@ -0,0 +1,573 @@
+            'edi_content': self._l10n_es_tbai_get_invoice_content_edi,
+        }
+
+    def _check_move_configuration(self, invoice):
+        # EXTENDS account_edi
+        errors = super()._check_move_configuration(invoice)
+
+        if self.code != 'es_tbai' or invoice.country_code != 'ES':
+            return errors
+
+        # Ensure a certificate is available.
+        if not invoice.company_id.l10n_es_edi_certificate_id:
+            errors.append(_("Please configure the certificate for TicketBAI/SII."))
+
+        # Ensure a tax agency is available.
+        if not invoice.company_id.mapped('l10n_es_tbai_tax_agency')[0]:
+            errors.append(_("Please specify a tax agency on your company for TicketBAI."))
+
+        # Check the refund reason
+        if invoice.move_type == 'out_refund':
+            if not invoice.l10n_es_tbai_refund_reason:
+                raise ValidationError(_('Refund reason must be specified (TicketBAI)'))
+            if invoice._is_l10n_es_tbai_simplified():
+                if invoice.l10n_es_tbai_refund_reason != 'R5':
+                    raise ValidationError(_('Refund reason must be R5 for simplified invoices (TicketBAI)'))
+            else:
+                if invoice.l10n_es_tbai_refund_reason == 'R5':
+                    raise ValidationError(_('Refund reason cannot be R5 for non-simplified invoices (TicketBAI)'))
+
+        return errors
+
+    def _l10n_es_tbai_post_invoice_edi(self, invoice):
+        # EXTENDS account_edi
+        if self.code != 'es_tbai':
+            return super()._post_invoice_edi(invoice)
+
+        # Chain integrity check: chain head must have been REALLY posted (not timeout'ed)
+        # - If called from a cron, then the re-ordering of jobs should prevent this from triggering
+        # - If called manually, then the user will see this error pop up when it triggers
+        chain_head = invoice.company_id._get_l10n_es_tbai_last_posted_invoice()
+        if chain_head and chain_head != invoice and not chain_head._l10n_es_tbai_is_in_chain():
+            raise UserError(f"TicketBAI: Cannot post invoice while chain head ({chain_head.name}) has not been posted")
+
+        # Generate the XML values.
+        inv_dict = self._get_l10n_es_tbai_invoice_xml(invoice)
+        if 'error' in inv_dict[invoice]:
+            return inv_dict  # XSD validation failed, return result dict
+
+        # Store the XML as attachment to ensure it is never lost (even in case of timeout error)
+        inv_xml = inv_dict[invoice]['xml_file']
+        invoice._update_l10n_es_tbai_submitted_xml(xml_doc=inv_xml, cancel=False)
+
+        # Assign unique 'chain index' from dedicated sequence
+        if not invoice.l10n_es_tbai_chain_index:
+            invoice.l10n_es_tbai_chain_index = invoice.company_id._get_l10n_es_tbai_next_chain_index()
+
+        # Call the web service and get response
+        # NOTE: we could store that response
```

Comment:
What is the point of this comment still?  @odoostan

Reply:
Nitpicking.